File: ui/src/main/java/org/wcl/mfn/ui/controller/HomeController.java
package org.wcl.mfn.ui.controller;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Controller;
import org.springframework.ui.Model;
import org.springframework.web.bind.annotation.GetMapping;
import org.wcl.mfn.ui.config.mvc.ViewConfig;

@Controller
public class HomeController {
    private final ViewConfig viewConfig;

    private HomeController(@Autowired ViewConfig viewConfig) {
        this.viewConfig = viewConfig;
    }

    @GetMapping("/")
    public String homePage(Model model) {
        model.addAttribute("page_title", "MFN Manager - Home");

        return viewConfig.homeView();
    }
}

File: ui/src/main/kotlin/org/wcl/mfn/ui/controller/HomeController.kt
package org.wcl.mfn.ui.controller

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.context.annotation.PropertySource
import org.springframework.stereotype.Controller
import org.springframework.ui.Model
import org.springframework.web.bind.annotation.GetMapping
import org.wcl.mfn.config.ui.common.PageTitleConfig
import org.wcl.mfn.config.ui.mvc.*

@Controller
@PropertySource("classpath:/url/url.properties")
class HomeController {
    @Autowired
    private val pageTitleConfig: PageTitleConfig? = null

    @Autowired
    private val modelConfig: ModelConfig? = null

    @Autowired
    private val viewConfig: ViewConfig? = null

    @GetMapping("#{'\${url.home}'}")
    fun homePage(model: Model): String? {
        modelConfig!!.pageTitle()?.let { model.addAttribute(it, pageTitleConfig!!.homePageTitle()) }

        return viewConfig!!.homeView()
    }
}
